### ISSUE TITLE
Commands blanked from cursor in REPL

### ISSUE THREAD

**Issue by jrmhaig** [mu-editor/mu#82]
When executing a command in the REPL, if enter is pressed whilst the cursor is not at the end of the line then the part of the line after the cursor is blanked. The command does, however, execute correctly.

**Comment by matt-land** [comment]
I investigated this, and added a trap on the new line characters in REPLPane.keyPressEvent:
if key == Qt.Key_Enter or key == Qt.Key_Return:
            return
With the trap, I successfully prevented the text stopped being garbled. I think this 'garbling' is from the round trip to the microbit, and not from mu.

I captured the serial communication to the microbit from the repl panel with this string: print("xx") [ and then backed the cursor up between the word 'print' and opening the parenthesis. ]
It looked like this:
<img width="192" alt="screen shot 2016-06-03 at 4 46 30 pm" src="https://cloud.githubusercontent.com/assets/4174972/15795899/bf685760-29aa-11e6-8bb7-1e35eb03793c.png">

The text being sent looks fine to me. The "\r" was the Enter/Return being sent. 

Since mu is opening a serial link, and that usually means displayed text is being echo'ed back, it seems pretty likely this is the microbit implementation, and not mu.
